cgroup2/manager.go

```diff
@@ -485,13 +481,13 @@ func (c *Manager) Delete() error {
 		if !os.IsNotExist(err) {
 			return err
 		}
-	} else {
-		threaded = cgType == Threaded
 	}
 
-	if threaded {
+	var tasks []uint64
+	switch cgType {
+	case Threaded, DomainThreaded:
 		tasks, err = c.Threads(true)
-	} else {
+	default:
 		tasks, err = c.Procs(true)
```

Comment:
Alternative could be to just to try  either if we get anything in return?

```go
if tasks, _ := c.Threads(true); len(tasks) > 0 {
	return fmt.Errorf("cgroups: unable to remove path %q: still contains running tasks", c.path)
}
if tasks, _ := c.Procs(true); len(tasks) > 0 {
	return fmt.Errorf("cgroups: unable to remove path %q: still contains running tasks", c.path)
}
```

Reply:
oh, not sure if that makes sense. Looking at `Procs` and `Threads`, they both call `getTasks` under the hood; if the above _does_ make sense, I guess that function could possibly accept  both `cgroupProcs`  and `cgroupThreads` to get "any".